src/test/java/redis/clients/jedis/commands/jedis/ScriptingCommandsTest.java

```diff
@@ -315,6 +319,133 @@ public void scriptExistsWithBrokenConnection() {
+    jedis.functionFlush();
+    assertEquals("OK", jedis.functionLoad(engine, library, function));
+    jedis.functionFlush(FlushMode.ASYNC);
+    assertEquals("OK", jedis.functionLoad(engine, library, function));
+    jedis.functionFlush(FlushMode.SYNC);
+  }
+
+  @Test
+  public void functionList() {
+    jedis.functionFlush();
+    String engine = "LUA";
+    String library = "mylib";
+    String function = "redis.register_function('myfunc', function(keys, args) return args[1] end)";
+    jedis.functionLoad(engine, library, new FunctionLoadParams().libraryDescription("test"), function);
+
+    LibraryInfo response = jedis.functionList().get(0);
+    assertEquals(library, response.getName());
+    assertEquals(engine, response.getEngine());
+    assertEquals("test", response.getDescription());
+    assertEquals(1, response.getFunctions().size());
+
+    // check function info
+    LibraryInfo.FunctionInfo func = response.getFunctions().get(0);
+    assertEquals("myfunc", func.getName());
+    assertNull(func.getDescription());
+    assertTrue(func.getFlags().isEmpty());
+
+    // check WITHCODE
+    response = jedis.functionListWithCode().get(0);
+    assertEquals(function, response.getCode());
+
+    // check with LIBRARYNAME
+    response = jedis.functionList(library).get(0);
+    assertEquals(library, response.getName());
+
+    // check with code and with LIBRARYNAME
+    response = jedis.functionListWithCode(library).get(0);
+    assertEquals(library, response.getName());
+    assertEquals(function, response.getCode());
+
+    // Binary
+    response = jedis.functionList(library.getBytes()).get(0);
+    assertEquals(library, response.getName());
+
+    response = jedis.functionListWithCode(library.getBytes()).get(0);
+    assertEquals(library, response.getName());
+  }
+
+  @Test
+  public void functionDumpRestore() {
+    jedis.functionFlush();
+    String engine = "Lua";
+    String library = "mylib";
+    String function = "redis.register_function('myfunc', function(keys, args) return args[1] end)";
+
+    jedis.functionLoad(engine, library, function);
+    byte[] payload = jedis.functionDump();
+    jedis.functionFlush();
+    assertEquals("OK", jedis.functionRestore(payload));
+    jedis.functionFlush();
+    assertEquals("OK", jedis.functionRestore(payload, FunctionRestorePolicy.FLUSH));
+    jedis.functionFlush();
+    assertEquals("OK", jedis.functionRestore(payload, FunctionRestorePolicy.APPEND));
+    jedis.functionFlush();
+    assertEquals("OK", jedis.functionRestore(payload, FunctionRestorePolicy.REPLACE));
+    jedis.functionFlush();
+  }
+
+  @Test
+  public void functionKill() {
+    jedis.functionFlush();
+    String engine = "Lua";
+    String library = "mylib";
+    String function = "redis.register_function('myfunc', function(keys, args)\n local a = 1 while true do a = a + 1 end \nend)";
+
+    jedis.functionLoad(engine, library, function);
+//    jedis.fcall("myfunc", new ArrayList<>(), new ArrayList<>());
```

Comment:
del unused code

Reply:
@zeekling She wants to keep this test and asked for help to make this test work...

> Can anyone help me to test FUNCTION KILL? The client is waiting to `fcall` reply so I can't execute `functionKill`...

Reply:
get it